File: testbed_ruby/test/associations/unidirectional_n_test.rb
require 'test/unit'
require 'src/student_a_k'
require 'src/mentor_a_k'
require 'src/program_a_k'

module CruiseTest
class UnidirectionalNTest < Test::Unit::TestCase

  def test_constructorEmpty
    m = MentorAK.new("blah")
    assert_equal(0,m.number_of_students)
  end
  
  def test_setTooMany
    s = StudentAK.new(99)
    s2 = StudentAK.new(98)
    s3 = StudentAK.new(97)
    s4 = StudentAK.new(96)
    
    m = MentorAK.new("blah")
    assert_equal(false,m.set_students([s,s2,s3,s4]))
    assert_equal(0, m.number_of_students)
  end

  def test_set_students
    s = StudentAK.new(99)
    s2 = StudentAK.new(98)
    s3 = StudentAK.new(97)
    s4 = StudentAK.new(97)
    
    m = MentorAK.new("blah")
    
    assert_equal(false,m.set_students([s,s2,s3,s4]))
    assert_equal(false,m.set_students([s,s2,s2]))
    assert_equal(true,m.set_students([s,s2]))
    
    assert_equal(2,m.number_of_students)
  end
  
  def test_deleteDoesNotChangeStudent
    s = StudentAK.new(99)
    s2 = StudentAK.new(98)
    s3 = StudentAK.new(97)

    p = ProgramAK.new
    s.set_program(p)
    
    m = MentorAK.new("blah")
    m.set_students([s,s2,s3])
    m.delete
    assert_equal(0,m.number_of_students)
    assert_equal(p,s.get_program)
  end
end
end

File: testbed_php/test/associations/UnidirectionalNTest.php
<?php

class UnidirectionalNTest extends UnitTestCase
{

  public function test_constructorEmpty()
  {
    $m = new MentorAK("blah");
    $this->assertEqual(0,$m->numberOfStudents());
  }
  
  public function test_setTooMany()
  {
    $s = new StudentAK(99);
    $s2 = new StudentAK(98);
    $s3 = new StudentAK(97);
    $s4 = new StudentAK(96);
    
    $m = new MentorAK("blah");
    $this->assertEqual(false,$m->setStudents(array($s,$s2,$s3,$s4)));
    $this->assertEqual(0, $m->numberOfStudents());
  }

  

  public function test_setStudents()
  {
    $s = new StudentAK(99);
    $s2 = new StudentAK(98);
    $s3 = new StudentAK(97);
    $s4 = new StudentAK(97);
    
    $m = new MentorAK("blah");
    
    $this->assertEqual(false,$m->setStudents(array($s,$s2,$s3,$s4)));
    $this->assertEqual(false,$m->setStudents(array($s,$s2,$s2)));
    $this->assertEqual(true,$m->setStudents(array($s,$s2)));
    
    $this->assertEqual(2,$m->numberOfStudents());
  }
  

  public function test_deleteDoesNotChangeStudent()
  {
    $s = new StudentAK(99);
    $s2 = new StudentAK(98);
    $s3 = new StudentAK(97);

    $p = new ProgramAK();
    $s->setProgram($p);
    
    $m = new MentorAK("blah");
    $m->setStudents(array($s,$s2,$s3));
    $m->delete();
    $this->assertEqual(0,$m->numberOfStudents());
    $this->assertEqual($p,$s->getProgram());
  }
}
